signing_service/src/signing_service.py

```diff
@@ -0,0 +1,139 @@
+            except KeyboardInterrupt:
+                # Handle keyboard interrupt.
+                self._terminate_connection_and_exit()
+            except Exception as exception:
+                # Try to close opened socket.
+                self.socket.close()
+                # If there was an unrecognized exception, log it and report to Sentry.
+                logger.error(f'Unrecognized exception occurred: {exception}')
+                crash_logger.error(str(exception))
+                raise
+
+    def _connect(self) -> None:
+        """ Creates socket and connects to given HOST and PORT. """
+        self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+        if self.current_reconnect_delay > 0:
+            logger.warning(f'Waiting {self.current_reconnect_delay} before connecting.')
+            sleep(self.current_reconnect_delay)
+
+        logger.warning(f'Connecting to {self.host}:{self.port}.')
+        self.socket.connect((self.host, self.port))
+        logger.warning(f'Connection established.')
+
+        # Reset delay on successful connection and set flag that connection is established.
+        self.current_reconnect_delay = 0
+        self.is_connection_established = True
+        self._handle_connection()
+
+    def _handle_connection(self) -> None:
+        """ Inner loop that handles data exchange over socket. """
+        while True:
+            data = self.socket.recv(1024)
+            self.socket.send(data)
+
+    def _terminate_connection_and_exit(self) -> None:
+        """ Close socket and exit. """
+        self.socket.close()
+        logger.info('Closing connection and exiting.')
+        exit()
+
+    def _increase_delay(self) -> None:
+        """ Increase current delay if connection cannot be established. """
+        if self.current_reconnect_delay == 0:
+            self.current_reconnect_delay = self.reconnect_delay
+        else:
+            self.current_reconnect_delay *= 2
+
+
+def _parse_arguments() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'concent_cluster_address',
+        help='Address of a Concent cluster.',
+    )
+    parser.add_argument(
+        'reconnect_delay',
+        type=int,
+        help='Delay between reconnections, doubles after each unsuccessful attempt and is reset after success.',
+    )
+
+    return parser.parse_args()
+
+
+if __name__ == '__main__':
+    logging.config.fileConfig(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', 'logging.ini'))
+
+    # Parse required arguments.
+    args = _parse_arguments()
+
+    try:
+        arg_host, arg_port = args.concent_cluster_address.split(':')
+    except ValueError:
+        exit('`concent_cluster_address` must be in format HOST:PORT.')
+
+    try:
+        arg_port = int(arg_port)
+    except ValueError:
+        exit('Port must be integer.')
+
+    arg_reconnect_delay = args.reconnect_delay
```

Comment:
this could be checked if it is convertible to `int` as well

Reply:
It is validated in 
```
parser.add_argument(
        'reconnect_delay',
        type=int,
        help='Delay between reconnections, doubles after each unsuccessful attempt and is reset after success.',
    )
```

Reply:
:+1: